pkg/cmd/runtime/plugins/k8s/k8s_runtime.go

```diff
@@ -21,3 +37,167 @@ func (p *K8sRuntime) Name() string { return "k8s" }
 func (p *K8sRuntime) Run(app string, args ...string) error {
 	return errors.New("TODO")
 }
+
+func (p *K8sRuntime) GetDefaultClient() (clientset *kubernetes.Clientset, err error) {
+	config, err := clientcmd.BuildConfigFromFlags("", homedir.HomeDir()+"/.kube/config")
+	if err != nil {
+		return
+	}
+	clientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (p *K8sRuntime) CreateCluster(appConf string, shouldWait bool, args ...string) (clusterId string, err error) {
+	if len(args) >= 1 {
+		clusterId = args[0]
+	} else {
+		clusterId = p.GetClusterId()
+	}
+	clientset, err := p.GetDefaultClient()
+	if err != nil {
+		logger.Errorf("Failed to get client when create cluster: %v", err)
+		return
+	}
+
+	decoder := scheme.Codecs.UniversalDeserializer()
+	d := yaml.NewYAMLOrJSONDecoder(strings.NewReader(appConf), 4096)
+	for {
+		ext := apiruntime.RawExtension{}
+		if err := d.Decode(&ext); err != nil {
+			if err == io.EOF {
+				err = nil
+				return clusterId, err
+			}
+			return clusterId, err
+		}
+		ext.Raw = bytes.TrimSpace(ext.Raw)
+		if len(ext.Raw) == 0 || bytes.Equal(ext.Raw, []byte("null")) {
+			continue
+		}
+		obj, _, err := decoder.Decode(ext.Raw, nil, nil)
+		switch o := obj.(type) {
+		case *v1beta1.Deployment:
+			o.SetName(p.GetName(clusterId, o.GetName()))
+			deploymentsClient := clientset.ExtensionsV1beta1().Deployments(v1.NamespaceDefault)
+			_, err = deploymentsClient.Create(o)
+			if err != nil {
+				logger.Errorf("Failed to create cluster: %v", err)
+				return clusterId, err
+			}
+		default:
+			logger.Errorf("Failed to create cluster with unknow type")
+			return clusterId, fmt.Errorf("Failed to create cluster [%s] with unknow type", clusterId)
+		}
+	}
+
+	return
+}
+
+func (p *K8sRuntime) StartClusters(clusterIds string, shouldWait bool, args ...string) error {
+	if len(args) < 1 {
```

Comment:
if `args[0]` must exists, the arguments should be `StartClusters(clusterIds string, shouldWait bool, appConf  string, ...)`

Reply:
Here implement runtime, k8s runtime need appConf, but qingcloud runtime not need this parameter